clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3104,20 +3104,58 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
       return;
   }
 
-  if (WorkGroupAttr *ExistingAttr = D->getAttr<WorkGroupAttr>()) {
-    ASTContext &Ctx = S.getASTContext();
+  ASTContext &Ctx = S.getASTContext();
+
+  if (!XDimExpr->isValueDependent() || !YDimExpr->isValueDependent() ||
+      !ZDimExpr->isValueDependent()) {
     Optional<llvm::APSInt> XDimVal = XDimExpr->getIntegerConstantExpr(Ctx);
     Optional<llvm::APSInt> YDimVal = YDimExpr->getIntegerConstantExpr(Ctx);
     Optional<llvm::APSInt> ZDimVal = ZDimExpr->getIntegerConstantExpr(Ctx);
-    Optional<llvm::APSInt> ExistingXDimVal = ExistingAttr->getXDimVal(Ctx);
-    Optional<llvm::APSInt> ExistingYDimVal = ExistingAttr->getYDimVal(Ctx);
-    Optional<llvm::APSInt> ExistingZDimVal = ExistingAttr->getZDimVal(Ctx);
 
-    // Compare attribute arguments value and warn for a mismatch.
-    if (ExistingXDimVal != XDimVal || ExistingYDimVal != YDimVal ||
-        ExistingZDimVal != ZDimVal) {
-      S.Diag(AL.getLoc(), diag::warn_duplicate_attribute) << AL;
-      S.Diag(ExistingAttr->getLocation(), diag::note_conflicting_attribute);
+    if (AL.getKind() == ParsedAttr::AT_ReqdWorkGroupSize) {
+      if (const auto *A = D->getAttr<SYCLIntelNumSimdWorkItemsAttr>()) {
+        int64_t NumSimdWorkItems =
+            A->getValue()->getIntegerConstantExpr(Ctx)->getSExtValue();
+
+        assert(NumSimdWorkItems && XDimVal &&
```

Comment:
I don't think these assertions make a lot of sense (we definitely don't need to assert `NumSimdWorkItems` multiple times). This is testing that the value comes back nonzero, which has nothing to do with whether the expr is an ICE.

Btw, one of the pushes @elizabethandrews and I are working on is to start converting all of these attributes to storing a `ConstantExpr` rather than an `Expr` so that these sort of uses become less awkward. I don't think you need to do that work as part of this patch, but I suspect these assertions would be going away under that model and so we can probably just drop them now unless you think they add some value that I'm not seeing.

Reply:
Done